controllers/operatorconfig_controller.go

```diff
@@ -180,10 +182,37 @@ func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		err := r.reconcileOperand(o, ctx, log, operatorConfig)
 		if err != nil {
 			log.Error(err, "Error creating resources")
+			if meta.FindStatusCondition(operatorConfig.Status.Conditions, ErrorCreatingResourcesConditionType) == nil {
```

Comment:
Looking at https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#SetStatusCondition ... why do we look for it first? It already has logic for adding it if it's not there?

Reply:
My bad!